engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -179,16 +193,25 @@ static Object doUncached(
   }
 
   @ExportMessage
-  Text toDisplayString(boolean allowSideEffects, @CachedLibrary("this") InteropLibrary atoms) {
+  Text toDisplayString(boolean allowSideEffects, @CachedLibrary("this") InteropLibrary atoms, @Cached BranchProfile handleError) {
+    Object result = null;
+    String msg;
     try {
-      return TypesGen.expectText(atoms.invokeMember(this, "to_text"));
-    } catch (UnsupportedMessageException
-        | ArityException
-        | UnknownIdentifierException
-        | UnsupportedTypeException
-        | UnexpectedResultException e) {
-      return Text.create(this.toString(10));
+      result = atoms.invokeMember(this, "to_text");
+      return TypesGen.expectText(result);
+    } catch (AbstractTruffleException panic) {
+      handleError.enter();
+      msg = this.toString("Panic in method `to_text` of [", 10, "]: ", panic);
+    } catch (UnexpectedResultException dataflowError) {
+      handleError.enter();
+      msg = this.toString("Error in method `to_text` of [", 10, "]: ", result);
+    } catch (
+      UnsupportedMessageException | ArityException | UnknownIdentifierException | UnsupportedTypeException e
+    ) {
+      handleError.enter();
+      msg = this.toString(null, 10, null, null);
```

Comment:
Why not do the `Panic in method` for these exceptions?

Reply:
It is panic since [44978cd](https://github.com/enso-org/enso/pull/4034/commits/44978cd0c59705bc662a75a1fd14db4cd83c9398)